packages/features/form-builder/schema.ts

```diff
@@ -318,6 +338,34 @@ export const fieldTypesSchemaMap: Partial<
       });
     },
   },
+  textarea: {
+    preprocess: ({ response }) => {
+      return response.trim();
+    },
+    superRefine: ({ field, response, ctx, m }) => {
+      const fieldTypeConfig = fieldTypesConfigMap[field.type];
+      const value = response ?? "";
+      const maxLength = field.maxLength ?? fieldTypeConfig.supportsLengthCheck?.maxLength;
+      if (!maxLength) {
+        throw new Error("maxLength must be there for textarea field");
+      }
+      const hasExceededMaxLength = value.length > maxLength;
+      const hasNotReachedMinLength = value.length < (field.minLength ?? 0);
+      if (hasExceededMaxLength) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: m(`Max. ${field.maxLength} characters allowed`),
+        });
+      }
```

Comment:
![image](https://github.com/user-attachments/assets/0057174e-8130-4b0d-af41-3aa7e7d15e7e)
Getting this error for a field that existed before these changes. Such a field doesn't have field.maxLength set